### PR TITLE
Closes #27, Closes #34 Mirror bootstrap build and deploy tooling

### LINKED ISSUE #27: Improve release management, and use github actions repository dispatch to open PR's on Quickstart upon new releases.
## Motivation
It is cumbersome and time consuming to create a new release for az_icons and populate it everywhere.

## Proposed Resolution
Using arizona-bootstrap's release management as a guide, implement the same thing for az-icons.

### LINKED ISSUE #34: Mirror arizona-bootstrap build tools, and deployment procedure.
## Motivation
Updating the az-icons cdn assets can be tedious, and prone to error

### Is your feature request related to a problem? Please describe.
Our arizona-bootstrap build and deployment process are much easier.

## Proposed Resolution
Mirror the build and deployment functionality found in arizona-bootstrap and supply some local build tools to cut down on manual errors.

### PR DESCRIPTION
**Pull request by trackleft**
## Description
This PR refactors our GitHub Actions workflows to better handle the release and deployment of az_quickstart. Significant changes include:
- Modifying `cdn-deploy-release.yml` to trigger on `repository_dispatch` with type `az_icons_release` instead of `workflow_dispatch`.
- Removing the creation of the release from `cdn-deploy-release.yml` and moving it to a new workflow `release.yml`.
- Updating the Dockerfile and scripts to accommodate these changes.
- Minor modifications to `CONTRIBUTING.md` and `COPYRIGHT.txt` to reflect the current year.
- Added Dockerfile
- Removal of an unused script reference in `demo.html`.
- Adjustments in `package.json` to refine the included files.

## Related issues
Closes #27
Closes #34 

## How to test 
**See my forks where I tested**:
- https://github.com/trackleft/az-icons/actions
- https://github.com/trackleft/az-icons-packagist/actions

The changes in this PR involve GitHub Actions workflows that interact between multiple repositories via repository dispatch events. Proper testing of these changes requires setting up your environment to mirror the interaction between the forks. Follow these steps:

1. Fork the following repositories to your own GitHub account:
   - `az-digital/az-icons`: The primary repository where the initial GitHub Action will run.
   - `az-digital/az-icons-packagist`: This repository is triggered by the `az-digital/az-icons` via a `repository_dispatch` event.

2. In your fork of the `az-digital/az-icons` repository, check out the branch associated with this PR `git checkout issue/34-mirror-arizona-bootstrap-build-tools`. This ensures you are testing the exact changes proposed.

3. Modify all workflow files in your fork that are part of this PR to reference your forked repositories instead of the original `az-digital` ones. Specifically, look for any `repository_dispatch` actions or other steps that reference `az-digital/az-icons` or `az-digital/az-icons-packagist` and update them to point to your forks, e.g., `username/az-icons` and `username/az-icons-packagist`.

4. Configure the necessary GitHub Secrets in each forked repository to allow the GitHub Actions to run properly. Specifically:
   - For `username/az-icons`:
     - `REPO_DISPATCH_TOKEN`: A personal access token (PAT) with the `repo` scope to trigger workflows in other repositories via repository dispatch. This token should have permissions for your forked repositories.
   - For `username/az-icons-packagist`:
     - Ensure any secrets needed for the actions triggered by the dispatch event are set, such as deployment keys or access tokens for external services. The exact secrets will depend on the actions defined in the workflow files.

5. Use the GitHub API or GitHub's UI in your forked `username/az-icons` repository to dispatch a `repository_dispatch` event with the type `az_icons_release`. This action simulates the trigger that the GitHub Actions workflow responds to.

6. Monitor the Actions tab in your forked `username/az-icons` repository to ensure the workflow is initiated correctly and behaves as expected.

7. Confirm that the `repository_dispatch` event triggers the expected actions in the `username/az-icons-packagist` repository and that they execute successfully.

8. Review the actions' outputs and execution logs in both repositories to ensure all steps are completed correctly, and any assets are properly deployed or released.

9. Based on the test results, if adjustments are needed, they should be made in the original PR branch, and the testing process repeated to confirm that the changes behave as intended.

After testing, remember to clean up any test releases, tags, or deployments in your forked repositories.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  workflow_dispatch:
+    inputs:
+      version:
+        description: The version to tag and release
+        required: true
+
+jobs:
+  release:
+    name: Create Release
+    runs-on: ubuntu-20.04
```
Comment (trackleft) [inline .github/workflows/release.yml:12]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (trackleft) [inline .github/workflows/release.yml:15]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+jobs:
+  release:
+    name: Create Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ssh-key: ${{ secrets.AZ_ICONS_DEPLOY_KEY }}
+
```
Comment (trackleft) [inline .github/workflows/release.yml:18]:
```suggestion
    steps:
      - name: Checkout code
        uses: actions/checkout@v4
        with:
          token: ${{ secrets.REPO_DISPATCH_TOKEN }}

```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+          imageprefix="${registry}/${GITHUB_REPOSITORY}/"
+          imagename='az-nodejs-ephemeral'
+          imagestem="${imageprefix}${imagename}:"
+          echo "AZ_DOCKER_REGISTRY=${registry}" >> ${GITHUB_ENV}
+          echo "AZ_OLD_HASH=${oldhash}" >> ${GITHUB_ENV}
+          echo "AZ_IMAGE_STEM=${imagestem}" >> ${GITHUB_ENV}
+          echo "AZ_ICONS_SOURCE_DIR=az-icons-src" >> ${GITHUB_ENV}
+          echo "AZ_ICONS_FROZEN_DIR=/azbuild/az-icons" >> ${GITHUB_ENV}
+
+      - name: Docker authentication
```
Comment (trackleft) [inline .github/workflows/release.yml:32]:
```suggestion
          echo "AZ_ICONS_FROZEN_DIR=/azbuild/az-icons" >> ${GITHUB_ENV}

      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v2

      - name: Docker authentication
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+          name: artifact
+          path: variables.json
+
+  dispatch:
+    needs: release
+    strategy:
+      matrix:
+        repo:
+          - az-digital/az-icons
+          - az-digital/az-icons-packagist
```
Comment (trackleft) [inline .github/workflows/release.yml:94]:
```suggestion
          - az-digital/az-icons-packagist

```

**Review by RyanDool — DISMISSED** [review]
I was unable to test this but in comparing bootstrap files to what you've created/updated seems logical and I didn't spot glaring errors.

File: .github/workflows/cdn-deploy-release.yml [hunk 6]
```diff
@@ -43,4 +29,5 @@ jobs:
       - name: Update 'latest' CDN assets to S3 + CloudFront
         run: |
           aws s3 sync --delete --cache-control max-age=691200  dist/. s3://${{ secrets.AZ_DIGITAL_CDN_BUCKET }}/lib/az-icons/latest/
-          aws cloudfront create-invalidation --distribution-id ${{ secrets.AZ_DIGITAL_CDN }} --paths /lib/az-icons/latest/*
+          aws cloudfront create-invalidation --distribution-id ${{
+          secrets.AZ_DIGITAL_CDN }} --paths /lib/az-icons/latest/*
```
Comment (trackleft) [inline .github/workflows/cdn-deploy-release.yml:33]:
```suggestion
          aws cloudfront create-invalidation --distribution-id ${{ secrets.AZ_DIGITAL_CDN }} --paths /lib/az-icons/latest/*
```

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,114 @@
+name: Create Release
+on:
+  workflow_dispatch:
```
Comment (trackleft) [inline .github/workflows/release.yml:3]:
```suggestion
name: Create Release
run-name: Create `${{ inputs.version }}` release from `${{ github.ref_name }}` branch
on:
  workflow_dispatch:
```

**Comment by trackleft** [comment]
I've forked this repo and az-icons-packagist in order to test this:
- https://github.com/trackleft/az-icons/actions
- https://github.com/trackleft/az-icons-packagist/actions

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        required: true
+jobs:
+  release:
+    name: Create Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ssh-key: ${{ secrets.AZ_ICONS_DEPLOY_KEY }}
```
Comment (trackleft) [inline .github/workflows/release.yml:17]:
```suggestion
          token: ${{ secrets.REPO_DISPATCH_TOKEN }}
```

**Review by joeparsons — APPROVED** [review]

**Comment by mmunro-ltrr** [comment]
Looks good, but I'll take a quick look at what would be required to start using Node.js 20 everywhere that older versions are referenced.

**Review by mmunro-ltrr — APPROVED** [review]

**Pull request merged**